contracts/README.md

```diff
@@ -0,0 +1,180 @@
+# Devnet contracts for development
+
+This folder contains Cairo and Solidity contracts for developement of Devnet.
+If you wish you to check specifically one of the two chains contracts, please check:
+1. `solidity` folder for Ethereum related contracts.
+2. `cairo` folder for Starknet related contracts, and example of how to work with starknet without
+   running a L1 node.
+
+## Full example of e2e testing with Anvil and Devnet
+
+### Setup of the nodes
+Please ensure that you've [anvil](https://book.getfoundry.sh/getting-started/installation) installed (or you can do the same with HardHat, but the commands here are done with anvil).
+```bash
+anvil
+```
+
+For Starknet, ensure you've Devnet compiled and running with the following params:
+```bash
+cargo run -- --seed 42
+```
+
+Now both nodes are running, Devnet for Starkne and Anvil for Ethereum.
+
+Then, **open two terminals**:
+* one for Starknet then:
+  ```bash
+  cd contracts/cairo
+  scarb build
+  ```
+
+* one for Ethereum then:
+  ```bash
+  cd contracts/solidity
+  # on first run you may need this
+  forge install
+  forge build
+  ```
+
+### Etherum setup
+1. Use Devnet postman endpoint to load the `MockStarknetMessaging` contract:
+```bash
+curl -H 'Content-Type: application/json' \
+     -d '{"networkUrl": "http://127.0.0.1:8545"}' \
+     http://127.0.0.1:5050/postman/load_l1_messaging_contract
+```
+```json
+{
+    "messageContractAddress":"0x5fbdb2315678afecb367f032d93f642f64180aa3"
+}
+```
+
+2. Deploy the `L1L2.sol` contract in order to receive/send messages from/to L2.
+```bash
+# Ethereum terminal (contracts/solidity)
+source .env
+forge script script/L1L2.s.sol:Deploy --broadcast --rpc-url $ETH_RPC_URL --silent
```

Comment:
```suggestion
forge script script/L1L2.s.sol:Deploy --broadcast --rpc-url $ETH_RPC_URL 
```
...to make the below output visible

Reply:
Updated: https://github.com/glihm/starknet-devnet-rs/tree/7dd5d6e0fe8f14aca40ebd562cc3c5d5e16ce873/contracts/l1-l2-messaging